pkg/framework/didresolver/api.go

```diff
@@ -72,6 +72,6 @@ type Opt func(opts *didResolverOpts)
 // WithDidMethod to add did method
 func WithDidMethod(id string, method DidMethod) Opt {
 	return func(opts *didResolverOpts) {
-		opts.didMethods[id] = method
+		opts.didMethods = append(opts.didMethods, didMethod{id: id, DidMethod: method})
```

Comment:
This makes using `WithDidMethod` awkward. The user now must specify a regex instead of just a DID method. Eg. the regex "rie" matches the string "aries" yet they clearly are not the same strings. To lock down the definition, the user is forced to provide "^rie$", which is counter intuitive.

One way to enable "catch-alls" is to invert control of the pattern matching such that the `DidMethod` is responsible for it. Add an `Accept(method string) bool` method to `DidMethod`, thereby freeing the user from such awkwardness and, incidentally, simplifying `WithDidMethod` by removing the `id` arg:

```go
resolver := didresolver.New(
    WithDidMethod(peer.NewDIDResolver(store)),
    WithDidMethod(universal.Resolver(url)),
    WithDidMethod(sidetree.Resolver(method, url)),
)
```

Reply:
I agree

Reply:
done